tldextract/cache.py

```diff
@@ -13,8 +15,59 @@
 _DID_LOG_UNABLE_TO_CACHE = False
 
 
+def get_pkg_unique_identifier():
+    """
+    Generate an identifier unique to the python version, tldextract version, and python instance
+
+    This will prevent interference between virtualenvs and issues that might arise when installing
+    a new version of tldextract
+    """
+    try:
+        # pylint: disable=import-outside-toplevel
+        from tldextract._version import version
+    except ImportError:
+        version = "dev"
+
+    tldextract_version = "tldextract-" + version
+    python_env_name = os.path.basename(sys.prefix)
+    # just to handle the edge case of two identically named python environments
+    python_binary_path_short_hash = hashlib.md5(sys.prefix.encode("utf-8")).hexdigest()[:6]
+    python_version = ".".join([str(v) for v in sys.version_info[:-1]])
+    identifier_parts = [
+        python_version,
+        python_env_name,
+        python_binary_path_short_hash,
+        tldextract_version
+    ]
+    pkg_identifier = "__".join(identifier_parts)
+
+    return pkg_identifier
+
+
+def get_cache_dir():
+    """
+    Get a cache dir that we have permission to write to
+
+    Try to follow the XDG standard, but if that doesn't work fallback to the package directory
+    """
+    try:
+        # http://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
+        xdg_cache_home = os.getenv("XDG_CACHE_HOME", None)
+        if xdg_cache_home is None:
+            user_home = os.getenv("HOME", None)
+            xdg_cache_home = os.path.join(user_home, ".cache")
+        pkg_id = get_pkg_unique_identifier()
+        cache_dir_default = os.path.join(xdg_cache_home, "python-tldextract", pkg_id)
+
+        return os.path.expanduser(os.environ.get("TLDEXTRACT_CACHE", cache_dir_default))
+    except TypeError:  # noqa
```

Comment:
What is this error? What is the `noqa` for? It is wrapping a lot of lines. Could the number of lines within the `try` be tighter?

Reply:
(If you delete the `try` entirely, do tests still pass?)

Reply:
I'll get rid of this. It's for trying to  join a string and None

Reply:
okay it's removed. not necessary in the version I just pushed